### PR TITLE
Allow append_images to set .icns scaled images

### PR DESCRIPTION
**Pull request by radarhere**
Suggestion for #2513 

To set the scaled images within `.icns`, include them in an `append_images` parameter, to match with the `save_all` functionality of other formats. The order of the images does not matter - which image is which can determined by their size.

### CODE REVIEW

**Comment by wiredfool** [comment]
LGTM

**Comment by hugovk** [comment]
@radarhere The merge of https://github.com/python-pillow/Pillow/pull/3004 has conflicted this one.

**Comment by radarhere** [comment]
@hugovk okay, rebased

File: Tests/test_file_icns.py [hunk 1]
```diff
@@ -36,6 +35,22 @@ def test_save(self):
         self.assertEqual(reread.size, (1024, 1024))
         self.assertEqual(reread.format, "ICNS")
 
+    @unittest.skipIf(sys.platform != 'darwin', "requires MacOS")
+    def test_save_append_images(self):
+        im = Image.open(TEST_FILE)
+
+        temp_file = self.tempfile("temp.icns")
+        providedIm = Image.new('RGBA', (32, 32), (255, 0, 0, 0))
```
Comment (hugovk) [inline Tests/test_file_icns.py]: Shall we follow conventions and name it something like `provided_im`?

File: src/PIL/IcnsImagePlugin.py [hunk 2]
```diff
@@ -310,6 +310,8 @@ def _save(im, fp, filename):
 
     # create the temporary set of pngs
     iconset = tempfile.mkdtemp('.iconset')
+    providedImages = {im.width:im for im in
```
Comment (hugovk) [inline src/PIL/IcnsImagePlugin.py]: Shall we follow conventions and name it something like `provided_images`?

File: Tests/test_file_icns.py [hunk 3]
```diff
@@ -22,8 +22,7 @@ def test_sanity(self):
         self.assertEqual(im.size, (1024, 1024))
         self.assertEqual(im.format, "ICNS")
 
-    @unittest.skipIf(sys.platform != 'darwin',
-                     "requires MacOS")
+    @unittest.skipIf(sys.platform != 'darwin', "requires MacOS")
```
Comment (hugovk) [inline Tests/test_file_icns.py]: "MacOS" -> "macOS"

File: Tests/test_file_icns.py [hunk 4]
```diff
@@ -36,6 +35,22 @@ def test_save(self):
         self.assertEqual(reread.size, (1024, 1024))
         self.assertEqual(reread.format, "ICNS")
 
+    @unittest.skipIf(sys.platform != 'darwin', "requires MacOS")
```
Comment (hugovk) [inline Tests/test_file_icns.py]: "MacOS" -> "macOS"

File: Tests/test_file_icns.py (hunk 3 above)
Comment (radarhere) [reply Tests/test_file_icns.py]: Okay, thanks, done.

File: Tests/test_file_icns.py (hunk 4 above)
Comment (radarhere) [reply Tests/test_file_icns.py]: Okay, thanks, done.

File: Tests/test_file_icns.py (hunk 1 above)
Comment (radarhere) [reply Tests/test_file_icns.py]: Okay, sure. I've made that change.

File: src/PIL/IcnsImagePlugin.py (hunk 2 above)
Comment (radarhere) [reply src/PIL/IcnsImagePlugin.py]: Okay, sure. I've made that change.

**Pull request merged**